models/person.go

```diff
@@ -217,3 +220,9 @@ func (u *Person) CanViewDashboard() bool {
 func (u *Person) CanChangeType(p *Publication) bool {
 	return u.CanEditPublication(p) && p.Status != "public"
 }
+
+func (p *Person) IsContributor(c []*Contributor) bool {
```

Comment:
likewise, this shouldn't be a method, a plain function will do

if it needs to be method, dataset/publication.HasContributor(p) makes more sense